src/frontend/data/websocket.ts

```diff
@@ -0,0 +1,29 @@
+import RobustWebSocket from 'altamoon-robust-websocket';
+
+import { addResource } from 'data/stores/generics';
+import { WS_ENPOINT } from 'settings';
+import { modelName } from 'types/models';
+import { UploadableObject, Video } from 'types/tracks';
+
+import { appData } from './appData';
+
+type WSMessageType = {
+  resource: UploadableObject;
+  type: modelName;
+};
+
+let videoWebsocket: RobustWebSocket;
+
+export const initVideoWebsocket = (video: Video) => {
```

Comment:
Why are we passing the whole `video` object and not only the `video.id` ? IMO it's more readable if only `id` is given because we instantly know what field of the `video` object this function is using

Reply:
Because we are working on a video. An `id ` is just a string. You can pass anything else than a video id and it's still right. Let say we change the signature by `initVideoWebsocket = (id: Video['id']) => {` calling it `initVideoWebsocket('foo');` is perfectly right.